raster/src/main/scala/geotrellis/raster/interpolation/InverseDistanceWeighting.scala

```diff
@@ -0,0 +1,139 @@
+      tile
+    } else {
+      val ewr2 = equalWeightRadius*equalWeightRadius
+
+      def idw(points: Seq[PointFeature[D]], x: Double, y: Double)(filter: Double => Boolean) = {
+        var sum = 0.0
+        var count = 0
+        var weightSum = 0.0
+        var sampleSum = 0.0
+        var sampleCount = 0
+
+        for (point <- points) {
+          val dX = x - point.geom.x
+          val dY = y - point.geom.y
+          val d2 = dX * dX + dY * dY
+
+          if (filter(d2)) {
+            if (d2 <= ewr2) {
+              sampleSum += point.data
+              sampleCount += 1
+            } else {
+              val w = 1 / d2
+              sum += point.data * w
+              weightSum += w
+              count += 1
+            }
+          }
+        }
+
+        if (sampleCount == 0) {
+          if (count == 0) {
+            doubleNODATA
+          } else {
+            f(sum / weightSum)
+          }
+        } else {
+          f(sampleSum / sampleCount)
+        }
+      }
+
+      radius match {
+        case Double.PositiveInfinity =>
+          cfor(0)(_ < rows, _ + 1) { row =>
+            cfor(0)(_ < cols, _ + 1) { col =>
+              val x = rasterExtent.gridColToMap(col)
+              val y = rasterExtent.gridRowToMap(row)
+              val value = idw(points, x, y)(d2 => true)
+
+              tile.setDouble(col, row, value)
+            }
+          }
+        case _ =>
+          val r2 = radius * radius
+          val index: SpatialIndex[PointFeature[D]] = SpatialIndex(points)(p => (p.geom.x, p.geom.y))
+
+          cfor(0)(_ < rows, _ + 1) { row =>
+            cfor(0)(_ < cols, _ + 1) { col =>
+              val x = rasterExtent.gridColToMap(col)
+              val y = rasterExtent.gridRowToMap(row)
+              val rPoints = index.pointsInExtent(Extent(x - radius, y - radius, x + radius, y + radius))
+              val value = if (rPoints.isEmpty) {
+                doubleNODATA
+              } else {
+                idw(rPoints, x, y)(d2 => d2 <= r2)
+              }
+
+              tile.setDouble(col, row, value)
+            }
+          }
+      }
+      tile
+    }
+  }
+
+
+  /**
+    * Compute an Inverse Distance Weighting raster over the given
+    * extent from the given set known-points.  Please see
+    * https://en.wikipedia.org/wiki/Inverse_distance_weighting for
+    * more details.
+    *
+    * @param   points    A collection of known-points
+    * @param   rasterExtent The study area
+    * @param   radius    Interpolation radius in cell units
+    * @param   equalWeightRadius If any points lie at that distance from a cell, the interpolation result for it would
+    *                            be the mean of those points
+    * @param   cellType  Interpolation radius in coordinate unit
+    * @param   f         Function to be applied before storing interpolation values
+    * @return            The data interpolated across the study area
+    */
+  def closerThan[D](
```

Comment:
I like the idea behind this overload but I don't think it belongs in the API. Here are some strikes against it:
 - The units of radius and equalWeightRadius are not the same
 - `Circle` is the only subtype of `Neighborhood` that is admirable here
 - `Circle` isn't actually being used as a cell mask, but is used for it's incidental property of `extent` and even then we have to use max of width or height
 - It's difficult to come up with a specific name for it.

I can imagine there could be a version of IDW where we actually used a cell `Neighborhood`, as in the actual cells mask where you could use instances other than Circle. But it would be a different implementation and it's not clear to me what use case it would really support.

At the moment I would suggest dropping this overload and getting the rest of the changes in, which already include a number of improvements and fixes.